src/app/modules/item/pages/item-log-view/item-log-view.component.ts

```diff
@@ -51,16 +40,51 @@ export class ItemLogViewComponent implements OnChanges {
   ) {}
 
   ngOnChanges(): void {
-    if (this.itemData) this.id$.next(this.itemData.item.id);
+    if (!this.itemData) {
+      return;
+    }
+
+    this.id$.next(this.itemData.item.id);
+    this.logColumns = this.getLogColumns(this.itemData.item.type);
   }
 
-  getData$(id: string): Observable<Data> {
+  private getData$(id: string): Observable<Data> {
     return this.activityLogService.getActivityLog(id).pipe(
       map((data: ActivityLog[]) => ({
-        columns: logColumns,
+        columns: this.logColumns || [],
         rowData: data
       }))
     );
   }
 
+  private getLogColumns(type: ItemType): Column[] {
+    const columns = [
```

Comment:
Or better, that satisfies both my previous remark:
Don't store `logColumns`, just call `getLogColumns` when you map your service. Anyway, your service is only called when you the item change, which is also the case when you "recompute" your columns.

(and BTW, I think there was an "error" if you would leave it like that: if your service returns immediately, the map may be executed before the columns are recomputed)

Reply:
Fixed